src/java/org/apache/cassandra/index/sai/disk/format/IndexFeatureSet.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.cassandra.index.sai.disk.format;
+
+/**
+ * The {@code IndexFeatureSet} represents the set of features available is SAI that are not
+ * available in the {@code V1OnDiskFormat}. Version 1 features should not be added to here
+ * unless the feature is removed from a future version and Version 1 is still supported.
+ */
+public interface IndexFeatureSet
+{
+    /**
```

Comment:
This interface looks weirdly empty. While I understand its purpose thanks to the documentation, I don't know how the features are going to be represented. I expected it would have a method returning a set of features (the name suggests it has Set semantics), but there is no such set anywhere here. What am I missing?

Reply:
The methods on here would return `boolean` values with `false` always taking precedence. Maybe `IndexFeatureSet` is a bad name and `IndexFeatures` would be better.